### DISCUSSION TITLE
What are the default problem matchers?

### DISCUSSION THREAD

**Discussion by jtbandes** [community/community#25670] [Actions]
<p>I’ve read about custom <a href="https://github.com/actions/toolkit/blob/2f164000dcd42fb08287824a3bc3030dbed33687/docs/problem-matchers.md" rel="noopener nofollow ugc">problem matchers</a>, but it seems like some program output (ESLint, for example) is already picked up as errors in check logs, and shows up as a per-line annotation in the PR changed files screen.</p>
<p><strong>Which problem matchers are enabled by default?</strong> Rather than going to the effort of writing a custom matcher, I’d like to tailor the output of my job to fit with existing matchers.</p>

**Comment by jsoref** [comment, marked as answer]
<p>Interesting question.</p>
<p>I’m not sure if this is active:</p><aside class="onebox githubblob" data-onebox-src="https://github.com/actions/runner/blob/be9632302ceef50bfb36ea998cea9c94c75e5d4d/src/Runner.Plugins/Repository/v1.1/GitSourceProvider.cs#L39-L49">
  <header class="source">

      <a href="https://github.com/actions/runner/blob/be9632302ceef50bfb36ea998cea9c94c75e5d4d/src/Runner.Plugins/Repository/v1.1/GitSourceProvider.cs#L39-L49" target="_blank" rel="noopener">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/actions/runner/blob/be9632302ceef50bfb36ea998cea9c94c75e5d4d/src/Runner.Plugins/Repository/v1.1/GitSourceProvider.cs#L39-L49" target="_blank" rel="noopener">actions/runner/blob/be9632302ceef50bfb36ea998cea9c94c75e5d4d/src/Runner.Plugins/Repository/v1.1/GitSourceProvider.cs#L39-L49</a></h4>



  <pre class="onebox">    <code class="lang-cs">
      <ol class="start lines" start="39" style="counter-reset: li-counter 38 ;">
          <li>""problemMatcher"": [</li>
          <li>    {</li>
          <li>        ""owner"": ""checkout-git"",</li>
          <li>        ""pattern"": [</li>
          <li>            {</li>
          <li>                ""regexp"": ""^(fatal|error): (.*)$"",</li>
          <li>                ""message"": 2</li>
          <li>            }</li>
          <li>        ]</li>
          <li>    }</li>
          <li>]</li>
      </ol>
    </code>
  </pre>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>My guess is that for most repositories, most of the time, this is active:</p><aside class="onebox githubblob" data-onebox-src="https://github.com/actions/checkout/blob/25a956c84d5dd820d28caab9f86b8d183aeeff3d/dist/problem-matcher.json#L2-L12">
  <header class="source">

      <a href="https://github.com/actions/checkout/blob/25a956c84d5dd820d28caab9f86b8d183aeeff3d/dist/problem-matcher.json#L2-L12" target="_blank" rel="noopener">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/actions/checkout/blob/25a956c84d5dd820d28caab9f86b8d183aeeff3d/dist/problem-matcher.json#L2-L12" target="_blank" rel="noopener">actions/checkout/blob/25a956c84d5dd820d28caab9f86b8d183aeeff3d/dist/problem-matcher.json#L2-L12</a></h4>



  <pre class="onebox">    <code class="lang-json">
      <ol class="start lines" start="2" style="counter-reset: li-counter 1 ;">
          <li>"problemMatcher": [</li>
          <li>    {</li>
          <li>        "owner": "checkout-git",</li>
          <li>        "pattern": [</li>
          <li>            {</li>
          <li>                "regexp": "^(fatal|error): (.*)$",</li>
          <li>                "message": 2</li>
          <li>            }</li>
          <li>        ]</li>
          <li>    }</li>
          <li>]</li>
      </ol>
    </code>
  </pre>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>if you use <a href="https://github.com/actions/setup-python">actions/setup-python</a>, you probably get this:</p><aside class="onebox githubblob" data-onebox-src="https://github.com/actions/setup-python/blob/a1121449a217580167611672a6c61b4cf314e0f2/.github/python.json#L2-L17">
  <header class="source">

      <a href="https://github.com/actions/setup-python/blob/a1121449a217580167611672a6c61b4cf314e0f2/.github/python.json#L2-L17" target="_blank" rel="noopener">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/actions/setup-python/blob/a1121449a217580167611672a6c61b4cf314e0f2/.github/python.json#L2-L17" target="_blank" rel="noopener">actions/setup-python/blob/a1121449a217580167611672a6c61b4cf314e0f2/.github/python.json#L2-L17</a></h4>



  <pre class="onebox">    <code class="lang-json">
      <ol class="start lines" start="2" style="counter-reset: li-counter 1 ;">
          <li>"problemMatcher": [</li>
          <li>    {</li>
          <li>        "owner": "python",</li>
          <li>        "pattern": [</li>
          <li>            {</li>
          <li>                "regexp": "^\\s*File\\s\\\"(.*)\\\",\\sline\\s(\\d+),\\sin\\s(.*)$",</li>
          <li>                "file": 1,</li>
          <li>                "line": 2</li>
          <li>            },</li>
          <li>            {</li>
          <li>                "regexp": "^\\s*raise\\s(.*)\\(\\'(.*)\\'\\)$",</li>
          <li>                "message": 2</li>
          <li>            }</li>
          <li>        ]</li>
          <li>    }</li>
          <li>]</li>
      </ol>
    </code>
  </pre>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>If you use <a href="https://github.com/actions/setup-node">setup-node</a>, you probably get this:</p><aside class="onebox githubblob" data-onebox-src="https://github.com/actions/setup-node/blob/5c355be17065acf11598c7a9bb47112fbcf2bbdc/.github/tsc.json#L2-L17">
  <header class="source">

      <a href="https://github.com/actions/setup-node/blob/5c355be17065acf11598c7a9bb47112fbcf2bbdc/.github/tsc.json#L2-L17" target="_blank" rel="noopener">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/actions/setup-node/blob/5c355be17065acf11598c7a9bb47112fbcf2bbdc/.github/tsc.json#L2-L17" target="_blank" rel="noopener">actions/setup-node/blob/5c355be17065acf11598c7a9bb47112fbcf2bbdc/.github/tsc.json#L2-L17</a></h4>



  <pre class="onebox">    <code class="lang-json">
      <ol class="start lines" start="2" style="counter-reset: li-counter 1 ;">
          <li>"problemMatcher": [</li>
          <li>    {</li>
          <li>        "owner": "tsc",</li>
          <li>        "pattern": [</li>
          <li>            {</li>
          <li>                "regexp": "^([^\\s].*)[\\(:](\\d+)[,:](\\d+)(?:\\):\\s+|\\s+-\\s+)(error|warning|info)\\s+TS(\\d+)\\s*:\\s*(.*)$",</li>
          <li>                "file": 1,</li>
          <li>                "line": 2,</li>
          <li>                "column": 3,</li>
          <li>                "severity": 4,</li>
          <li>                "code": 5,</li>
          <li>                "message": 6</li>
          <li>            }</li>
          <li>        ]</li>
          <li>    }</li>
          <li>]</li>
      </ol>
    </code>
  </pre>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>and this:</p><aside class="onebox githubblob" data-onebox-src="https://github.com/actions/setup-node/blob/5c355be17065acf11598c7a9bb47112fbcf2bbdc/.github/eslint-compact.json#L2-L17">
  <header class="source">

      <a href="https://github.com/actions/setup-node/blob/5c355be17065acf11598c7a9bb47112fbcf2bbdc/.github/eslint-compact.json#L2-L17" target="_blank" rel="noopener">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/actions/setup-node/blob/5c355be17065acf11598c7a9bb47112fbcf2bbdc/.github/eslint-compact.json#L2-L17" target="_blank" rel="noopener">actions/setup-node/blob/5c355be17065acf11598c7a9bb47112fbcf2bbdc/.github/eslint-compact.json#L2-L17</a></h4>



  <pre class="onebox">    <code class="lang-json">
      <ol class="start lines" start="2" style="counter-reset: li-counter 1 ;">
          <li>"problemMatcher": [</li>
          <li>    {</li>
          <li>        "owner": "eslint-compact",</li>
          <li>        "pattern": [</li>
          <li>            {</li>
          <li>                "regexp": "^(.+):\\sline\\s(\\d+),\\scol\\s(\\d+),\\s(Error|Warning|Info)\\s-\\s(.+)\\s\\((.+)\\)$",</li>
          <li>                "file": 1,</li>
          <li>                "line": 2,</li>
          <li>                "column": 3,</li>
          <li>                "severity": 4,</li>
          <li>                "message": 5,</li>
          <li>                "code": 6</li>
          <li>            }</li>
          <li>        ]</li>
          <li>    }</li>
          <li>]</li>
      </ol>
    </code>
  </pre>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<p>
and this:</p><aside class="onebox githubblob" data-onebox-src="https://github.com/actions/setup-node/blob/5c355be17065acf11598c7a9bb47112fbcf2bbdc/.github/eslint-stylish.json#L2-L21">
  <header class="source">

      <a href="https://github.com/actions/setup-node/blob/5c355be17065acf11598c7a9bb47112fbcf2bbdc/.github/eslint-stylish.json#L2-L21" target="_blank" rel="noopener">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/actions/setup-node/blob/5c355be17065acf11598c7a9bb47112fbcf2bbdc/.github/eslint-stylish.json#L2-L21" target="_blank" rel="noopener">actions/setup-node/blob/5c355be17065acf11598c7a9bb47112fbcf2bbdc/.github/eslint-stylish.json#L2-L21</a></h4>



  <pre class="onebox">    <code class="lang-json">
      <ol class="start lines" start="2" style="counter-reset: li-counter 1 ;">
          <li>"problemMatcher": [</li>
          <li>    {</li>
          <li>        "owner": "eslint-stylish",</li>
          <li>        "pattern": [</li>
          <li>            {</li>
          <li>                "regexp": "^([^\\s].*)$",</li>
          <li>                "file": 1</li>
          <li>            },</li>
          <li>            {</li>
          <li>                "regexp": "^\\s+(\\d+):(\\d+)\\s+(error|warning|info)\\s+(.*)\\s\\s+(.*)$",</li>
          <li>                "line": 1,</li>
          <li>                "column": 2,</li>
          <li>                "severity": 3,</li>
          <li>                "message": 4,</li>
          <li>                "code": 5,</li>
          <li>                "loop": true</li>
          <li>            }</li>
          <li>        ]</li>
          <li>    }</li>
          <li>]</li>
      </ol>
    </code>
  </pre>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<p>
via:</p><aside class="onebox githubblob" data-onebox-src="https://github.com/actions/setup-node/blob/25316bbc1f10ac9d8798711f44914b1cf3c4e954/src/main.ts#L59-L64">
  <header class="source">

      <a href="https://github.com/actions/setup-node/blob/25316bbc1f10ac9d8798711f44914b1cf3c4e954/src/main.ts#L59-L64" target="_blank" rel="noopener">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/actions/setup-node/blob/25316bbc1f10ac9d8798711f44914b1cf3c4e954/src/main.ts#L59-L64" target="_blank" rel="noopener">actions/setup-node/blob/25316bbc1f10ac9d8798711f44914b1cf3c4e954/src/main.ts#L59-L64</a></h4>



  <pre class="onebox">    <code class="lang-ts">
      <ol class="start lines" start="59" style="counter-reset: li-counter 58 ;">
          <li>core.info(`##[add-matcher]${path.join(matchersPath, 'tsc.json')}`);</li>
          <li>core.info(</li>
          <li>  `##[add-matcher]${path.join(matchersPath, 'eslint-stylish.json')}`</li>
          <li>);</li>
          <li>core.info(</li>
          <li>  `##[add-matcher]${path.join(matchersPath, 'eslint-compact.json')}`</li>
      </ol>
    </code>
  </pre>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>and <a href="https://github.com/actions/setup-go">setup-go</a>:</p><aside class="onebox githubblob" data-onebox-src="https://github.com/actions/setup-go/blob/3b4dc6cbed1779f759b9c638cb83696acea809d1/matchers.json#L2-L15">
  <header class="source">

      <a href="https://github.com/actions/setup-go/blob/3b4dc6cbed1779f759b9c638cb83696acea809d1/matchers.json#L2-L15" target="_blank" rel="noopener">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/actions/setup-go/blob/3b4dc6cbed1779f759b9c638cb83696acea809d1/matchers.json#L2-L15" target="_blank" rel="noopener">actions/setup-go/blob/3b4dc6cbed1779f759b9c638cb83696acea809d1/matchers.json#L2-L15</a></h4>



  <pre class="onebox">    <code class="lang-json">
      <ol class="start lines" start="2" style="counter-reset: li-counter 1 ;">
          <li>"problemMatcher": [</li>
          <li>    {</li>
          <li>        "owner": "go",</li>
          <li>        "pattern": [</li>
          <li>            {</li>
          <li>                "regexp": "^\\s*(\\.{0,2}[\\/\\\\].+\\.go):(?:(\\d+):(\\d+):)? (.*)",</li>
          <li>                "file": 1,</li>
          <li>                "line": 2,</li>
          <li>                "column": 3,</li>
          <li>                "message": 4</li>
          <li>            }</li>
          <li>        ]</li>
          <li>    }</li>
          <li>]</li>
      </ol>
    </code>
  </pre>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Results extracted from:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/search?q=org:actions%2Bproblemmatcher&amp;type=code">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/search?q=org:actions%2Bproblemmatcher&amp;type=code" target="_blank" rel="noopener">GitHub</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/2119212/181092445-63227baf-e758-42f4-a192-991d7db7648b.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://github.com/search?q=org:actions%2Bproblemmatcher&amp;type=code" target="_blank" rel="noopener">Build software better, together</a></h3>

  <p>GitHub is where people build software. More than 65 million people use GitHub to discover, fork, and contribute to over 200 million projects.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
